admin/spam/views.py

```diff
@@ -1,9 +1,117 @@
-from django.http import HttpResponse
+import operator
 
+from django.shortcuts import render, render_to_response
+from django.core.paginator import Paginator, EmptyPage, PageNotAnInteger
+from django.contrib.auth.decorators import login_required
+from django.core.mail import send_mail
+from django.views.generic import FormView
+from django.core.urlresolvers import reverse
 
+from modularodm import Q
+from website.project.model import Comment
+
+from .serializers import serialize_comment
+from .forms import EmailForm
+
+
+class EmailFormView(FormView):
+
+    form_class = EmailForm
+    template_name = "spam/email.html"
+    spam_id = None
+    page = 1
+
+    def get(self, request, *args, **kwargs):
+        spam_id = kwargs.get('spam_id', None)
+        self.spam_id = spam_id
+        self.page = request.GET.get('page', 1)
+        try:
+            spam = serialize_comment(Comment.load(spam_id))
+        except:
+            pass
+        self.set_initial(spam)
+        form = self.get_form()
+        context = {
+            'comment': spam,
+            'page_number': request.GET.get('page', 1),
+            'form': form
+        }
+        return render(request, self.template_name, context)
+
+    def post(self, request, *args, **kwargs):
+        spam_id = kwargs.get('spam_id', None)
+        self.spam_id = spam_id
+        self.page = request.GET.get('page', 1)
+        try:
+            spam = serialize_comment(Comment.load(spam_id))
+        except:
+            pass
+        self.set_initial(spam)
+        form = self.get_form()
+        if form.is_valid():
+            return self.form_valid(form)
+        else:
+            return self.form_invalid(form)
+
+    def set_initial(self, spam):
+        self.initial = {
+            'author': spam['author'].fullname,
+            'email': [(r, r) for r in spam['author'].emails],
+            'subject': 'Reports of spam',
+            'message': render_to_response(
+                'spam/email_template.html',
+                {'item': spam}
+            ).content,
+        }
+
+    def form_valid(self, form):
+        send_mail(
+            subject=form.cleaned_data.get('subject').strip(),
+            message=form.cleaned_data.get('message'),
+            from_email='support@osf.io',
+            recipient_list=[form.cleaned_data.get('email')]
+        )
+        return super(EmailFormView, self).form_valid(form)
+
+    @property
+    def success_url(self):
+        return reverse('spam:detail', kwargs={'spam_id': self.spam_id}) + '?page={}'.format(self.page)
+
+
+def get_spam_list():
+    query = (
+        Q('reports', 'ne', {}) &
+        Q('reports', 'ne', None)
+    )
+    return sorted(
```

Comment:
Since `date_created` is a field in mongo you should be able to get mongo to sort the data instead of relying on python.
**Descending**
```python
return Comment.find(query).sort('-date_created')
```
**Ascending**
```python
return Comment.find(query).sort('date_created')
```